uiowa/uiowa
# Metatag Enhancements and Fixes

Resolves https://github.com/uiowa/uiowa/issues/2169
Resolves https://github.com/uiowa/uiowa/issues/2978
Resolves https://github.com/uiowa/uiowa/issues/3074

To make this possible, I had to establish the token view mode for the default content types. If layout builder is used and the featured image is not displayed, the token is null. This solves this and other situations.

## To Test

- v2/v3, admissions and other splits should have clean imports. Other sites touched include:
  - admissions, brand, policy.clas, ighn.international, grad, uiowa.edu, hr, presidentialsearch
- As a webmaster on a v2/v3 and any of the custom sites where config has been updated, you can see and set Twitter card metatag information (title, summary, image). Defaults are given.
- Inspecting the source you should see twitter metatag information for:
  - twitter:card which should be `summary_large_image`
  - twitter:description
  - twitter:title
  - twitter:image
- The Twitter Card fieldset is collapsed by default within the SEO settings for sites using common node add/edit defaults similar to Advanced and Open Graph.
- There are no notice notifications when add/editing content or metatag configuration as an Admin.

_Note: validating using twitter's https://cards-dev.twitter.com/validator will have to wait till stage or prod._

## Examples

**Regardless of website, as a webmaster you should see this on Page, Article, Person content type add/edit forms within the SEO Settings fieldset (collapsed by default):**
<img width="572" alt="Screen Shot 2021-03-26 at 11 51 57 AM" src="https://user-images.githubusercontent.com/4663676/112666190-e7482380-8e29-11eb-96a6-9c3e090a8a86.png">

**Regardless of website, you should see this kind of output in the source for Page, Article, Person assuming there is a Title, Featured Image/Person photo and a Content Description/Article Summary/Short Biography.**
<img width="420" alt="Screen Shot 2021-03-26 at 11 55 23 AM" src="https://user-images.githubusercontent.com/4663676/112666601-5a519a00-8e2a-11eb-8ff9-eb4da422c63d.png">
